### PR TITLE
Herding

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/usaco-918.mdx [hunk 1]
```diff
@@ -2,178 +2,168 @@
...
-
-		public InputReader() {
-			try {
-				reader = new BufferedReader(new InputStreamReader(System.in), 32768);
-			} catch (Exception e) {
-				throw new NullPointerException("Could not create input stream");
-			}
-		}
+public class Herding {
+	public static void main(String[] args) throws IOException{
```
Comment (ChuangSheep) [inline solutions/usaco-918.mdx:83]:
```suggestion
	public static void main(String[] args) throws IOException {
```

File: solutions/usaco-918.mdx [hunk 2]
```diff
@@ -2,178 +2,168 @@
...
+		BufferedReader br = new BufferedReader(new FileReader("herding.in"));
```
Comment (ChuangSheep) [inline solutions/usaco-918.mdx:84]: have u closed it?

Comment (SansPapyrus683) [reply solutions/usaco-918.mdx:84]: no? shouldn't matter

Comment (ChuangSheep) [reply solutions/usaco-918.mdx:84]: it indeed does not matter, it is just not a good practice leaving resources opened
https://stackoverflow.com/questions/54888904/should-i-always-close-bufferedreader

**Review by ChuangSheep — APPROVED** [review]
:blob_heart:

**Comment by SansPapyrus683** [comment]
@ryanchou-dev

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**